sweeper.go

```diff
@@ -162,10 +160,15 @@ func (sw *sweeper) update() {
 		m.snapshot.Total = total
 	}
 
-	// trim the meter list
-	for i := newLen; i < len(sw.meters); i++ {
-		sw.meters[i] = nil
+	// compress and trim the meter list
+	var newLen int
+	for _, m := range sw.meters {
+		if m != nil {
+			sw.meters[newLen] = m
+			newLen++
+		}
```

Comment:
There's actually a trick to this:

```go
newMeters := sw.meters[:0]
for _, m := range sw.meters {
    if m != nil {
        newMeters = append(newMeters, m)
    }
}
sw.meters = newMeters
```

Reply:
neat, but it probably generates the same code underneath :)

Reply:
do you want me to apply it? I am neutral either way and I usually prefer explicitness.

Reply:
actually, the trick might generate slower code because it has to do bounds checking. 
Maybe the compiler is smart enough to elide it though, I wouldn't bet against it.

Reply:
The trick will do _less_ bounds checking, IIRC. But I'm fine with this as it is.